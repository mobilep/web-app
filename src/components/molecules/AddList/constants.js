export const keyMap = {
	ENTER: 13,
	UP_ARROW: 38,
	DOWN_ARROW: 40,
	TAB: 9,
};

export default {
	keyMap,
};
