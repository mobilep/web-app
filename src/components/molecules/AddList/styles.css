.AddList {
	position: relative;
}

.AddList-item {
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: 4px 0;
}

.AddList-removeButton {
	display: flex;
	align-items: center;
	justify-content: center;
	width: 24px;
	height: 24px;
	padding: 0;
	border: none;
	color: var(--almostBlack);
	cursor: pointer;
	margin-right: 12px;
}

.AddList-trashIcon {
	width: 20px;
	height: 20px;
}

.AddList-addItem {
	margin-top: 4px;
}

.AddList-autoComplete {
	position: absolute;
	left: 29px;
	top: 100%;
}

.AddList-selectList {
	display: none;
}

.AddList-selectList:empty {
	display: none !important;
}

.AddList-activeHint .AddList-selectList,
.AddList-autoComplete:hover .AddList-selectList {
	display: block;
}

.AddList-selectList:hover:empty {
	display: none;
}
