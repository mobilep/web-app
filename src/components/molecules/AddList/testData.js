export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
