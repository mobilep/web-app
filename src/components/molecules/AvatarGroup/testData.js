export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
