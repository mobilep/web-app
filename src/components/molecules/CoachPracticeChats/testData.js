/* eslint-disable max-len */
export const practiceChats = [
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618100',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5d70bc6d93b6e912b83a65d5',
			firstName: 'Taras',
			lastName: 'Kvas',
			avatarId: '2e3db607-9ec1-45f4-a7e5-4bed03912dd5',
			devices: 'Array[0]',
			avatarColor: 'fd524f',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			havePassword: false,
			name: 'Taras Kvas',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.554Z',
		updatedAt: '2019-10-29T09:19:42.764Z',
		userEvaluatedAt: null,
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: true,
		isActive: true,
		lastMessageFrom: 'learner',
		status: 'current',
		coachMark: [],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 4,
		userAudiosCount: 0,
		userPhotosCount: 1,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 19,
		_inbox: {
			_id: '5dad8ef546396073dd36390c',
			updatedAt: '2019-10-29T09:20:28.648Z',
			createdAt: '2019-10-21T10:56:53.544Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5d70bc6d93b6e912b83a65d5',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618100/5dad8ef546396073dd36390c',
			lastReadMessage: '-LsLo4OhHveQpLP5JXKy',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: '2019-10-28T13:25:34.142Z',
			firstMessageAt: '2019-10-28T13:09:17.159Z',
			isActive: true,
		},
		state: 'waitingOnFeedback',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618102',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5da86a3716973b29347321ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			devices: 'Array[0]',
			avatarColor: '00cdbb',
			lang: 'en',
			avatar_sm: null,
			avatar_md: null,
			avatar_lg: null,
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.563Z',
		updatedAt: '2019-10-29T09:18:36.340Z',
		userEvaluatedAt: '2019-10-29T09:16:14.949Z',
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: false,
		isActive: true,
		lastMessageFrom: 'coach',
		coachMark: [
			'Object',
		],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 2,
		userAudiosCount: 0,
		userPhotosCount: 0,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 0,
		_inbox: {
			_id: '5dad8ef546396073dd36390d',
			updatedAt: '2019-10-29T09:19:11.848Z',
			createdAt: '2019-10-21T10:56:53.547Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5da86a3716973b29347321ce',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618102/5dad8ef546396073dd36390d',
			lastReadMessage: '-LsLnpAfH4s1IOZsQrgY',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: null,
			firstMessageAt: '2019-10-29T09:18:27.588Z',
			isActive: true,
		},
		state: 'completed',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
];

export const activeChatId = practiceChats[0]._id;
