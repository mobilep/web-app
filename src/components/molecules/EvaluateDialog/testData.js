export const evaluationList = [
	{
		id: '1',
		title: 'Contactâ€™s attention grabbed',
	}, {
		id: '2',
		title: 'Self Introduction',
	}, {
		id: '3',
		title: 'Purpose of call stated',
	},
];
