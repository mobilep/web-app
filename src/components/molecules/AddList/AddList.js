import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { AddItem, UserLabel, UserSelectList } from '../../atoms';
import { TrashIcon } from '../../../icons';
import { keyMap } from './constants';

import './styles.css';

const defaultState = {
	isAutoCompleteOpen: false,
	addUserValue: '',
	currentItem: null,
	selectedAutoCompleteItem: -1,
	currentAutoCompleteItems: [],
};

export default class AddList extends Component {

	state = defaultState;

	handleRemoveClick = (id) => () => this.props.onItemRemove(id);

	handleInputBlur = () => this.setState({ isAutoCompleteOpen: false });

	handleAddItemChange = ({ target: { value } }) => {
		this.setState({
			addUserValue: value,
			isAutoCompleteOpen: true,
			currentItem: null,
			selectedAutoCompleteItem: -1,
			currentAutoCompleteItems: this.getAutoCompleteItems(value),
		});
	}

	handleKeyDown = (event) => {
		if (!this.state.currentAutoCompleteItems) return false;

		switch (event.keyCode) {
			case keyMap.DOWN_ARROW:
				this.handleDownArrow();
				event.preventDefault();
				break;
			case keyMap.UP_ARROW:
				this.handleUpArrow();
				event.preventDefault();
				break;
			case keyMap.ENTER:
				this.handleEnter();
				event.preventDefault();
				break;
			default:
				return false;
		}
	}

	handleDownArrow () {
		const { selectedAutoCompleteItem, currentAutoCompleteItems } = this.state;

		if (selectedAutoCompleteItem < currentAutoCompleteItems.length - 1) {
			this.setState({ selectedAutoCompleteItem: selectedAutoCompleteItem + 1 });
		}
	}

	handleUpArrow () {
		const { selectedAutoCompleteItem } = this.state;

		if (selectedAutoCompleteItem >= -1) this.setState({ selectedAutoCompleteItem: selectedAutoCompleteItem - 1 });
	}

	handleEnter () {
		const { selectedAutoCompleteItem, currentAutoCompleteItems } = this.state;

		if (selectedAutoCompleteItem >= 0) {
			this.handleItemSelect(currentAutoCompleteItems[selectedAutoCompleteItem].id);
		} else {
			this.handleUserAdd();
		}
	}

	handleUserAdd () {
		const { onItemAdd, allItems } = this.props;
		const { currentItem, addUserValue } = this.state;

		if (currentItem) {
			onItemAdd(currentItem.id);
			this.setState(defaultState);
		} else {
			const item = allItems.find((item) => {
				return item.fullName.toLowerCase() === addUserValue.trim().toLowerCase();
			});

			if (item) {
				onItemAdd(item.id);
				this.setState(defaultState);
			}
		}

		return false;
	}

	getItems (items) {

		return items.map((item, index) => {
			const { id, fullName, initials } = item;
			const key = `item-${index}`;

			return (
				<div key={key} className='AddList-item'>
					<UserLabel
						fullName={fullName}
						initials={initials}
					/>

					<button onClick={this.handleRemoveClick(id)} className='AddList-removeButton'>
						<TrashIcon className='AddList-trashIcon' />
					</button>
				</div>
			);
		});
	}

	getAutoCompleteItems = (addUserValue) => {
		const { allItems } = this.props;

		if (!addUserValue) return [];
		else return allItems.filter((item) => item.fullName.toLowerCase().includes(addUserValue.toLowerCase()));
	}

	handleItemSelect = (id) => {
		const { currentAutoCompleteItems } = this.state;
		const currentItem = currentAutoCompleteItems.find((item) => item.id === id);

		this.setState({
			...defaultState,
			currentItem,
			addUserValue: currentItem.fullName,
		});
	}

	getClassName (className, isAutoCompleteOpen) {
		return classNames('AddList', {
			'AddList-activeHint': isAutoCompleteOpen,
		}, className);
	}

	render () {
		const { items, className } = this.props;
		const { isAutoCompleteOpen, selectedAutoCompleteItem, addUserValue, currentAutoCompleteItems } = this.state;

		return (
			<div className={this.getClassName(className, isAutoCompleteOpen)}>
				{this.getItems(items)}
				<AddItem
					className='AddList-addItem'
					placeholder='Add a few criteriaâ€¦'
					onBlur={this.handleInputBlur}
					onChange={this.handleAddItemChange}
					onKeyDown={this.handleKeyDown}
					value={addUserValue}
				/>

				<div className='AddList-autoComplete'>
					<UserSelectList
						className='AddList-selectList'
						items={currentAutoCompleteItems}
						selectedItem={selectedAutoCompleteItem}
						onSelect={this.handleItemSelect}
					/>
				</div>

			</div>
		);
	}
}

AddList.propTypes = {
	onItemAdd: () => {},
	onItemRemove: () => {},
};

AddList.propTypes = {
	allItems: PropTypes.array,
	className: PropTypes.string,
	items: PropTypes.array,
	onItemAdd: PropTypes.func,
	onItemRemove: PropTypes.func,
};
