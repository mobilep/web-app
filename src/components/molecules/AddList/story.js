import React from 'react';
import { storiesOf } from '@storybook/react';
import { withState } from '@dump247/storybook-state';

import AddList from './AddList';
import { allUsers, users } from './testData';

const props = { allUsers, users };

storiesOf('2. Molecules / AddList', module)
	.add('with static list', withState(props)(({ store }) => {

		const onItemRemove = (id) => {
			props.users = [...props.users].filter((user) => (user.id !== id));
			store.set({ users: props.users });
		};

		const onItemAdd = (id) => {
			props.allUsers = [...props.allUsers].filter((user) => {
				if (user.id === id) {
					props.users.push(user);

					store.set({ users: props.users });
				}

				return user.id !== id;
			});

			store.set({ allUsers: props.allUsers });
		};

		return (
			<AddList
				items={store.state.users}
				allItems={store.state.allUsers}
				onItemRemove={onItemRemove}
				onItemAdd={onItemAdd}
			/>
		);
	}));
