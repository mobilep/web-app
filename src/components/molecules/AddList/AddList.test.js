import React from 'react';
import { shallow } from 'enzyme';

import AddList from './AddList';
import { allUsers, users } from './testData';

describe('Molecule - AddList', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<AddList
				allItems={allUsers}
				items={users}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
