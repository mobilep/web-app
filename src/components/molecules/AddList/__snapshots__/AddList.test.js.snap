// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - AddList Should render 1`] = `
<div
  className="AddList"
>
  <div
    className="AddList-item"
    key="item-0"
  >
    <UserLabel
      fullName="John Smith"
      initials="JS"
    />
    <button
      className="AddList-removeButton"
      onClick={[Function]}
    >
      <TrashIcon
        className="AddList-trashIcon"
      />
    </button>
  </div>
  <div
    className="AddList-item"
    key="item-1"
  >
    <UserLabel
      fullName="Philippe Starck"
      initials="PS"
    />
    <button
      className="AddList-removeButton"
      onClick={[Function]}
    >
      <TrashIcon
        className="AddList-trashIcon"
      />
    </button>
  </div>
  <div
    className="AddList-item"
    key="item-2"
  >
    <UserLabel
      fullName="Suzy Salmon"
      initials="SS"
    />
    <button
      className="AddList-removeButton"
      onClick={[Function]}
    >
      <TrashIcon
        className="AddList-trashIcon"
      />
    </button>
  </div>
  <AddItem
    className="AddList-addItem"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Add a few criteriaâ€¦"
    value=""
  />
  <div
    className="AddList-autoComplete"
  >
    <UserSelectList
      className="AddList-selectList"
      items={Array []}
      onMouseOver={[Function]}
      onSelect={[Function]}
      selectedItem={-1}
    />
  </div>
</div>
`;
